src/tools/web/client/src/reducers/kdb.js

```diff
@@ -38,6 +38,13 @@ export default function keyReducer (state = {}, action) {
     case SET_KEY_REQUEST:
       return updateState(state, action.request)
 
+    // TODO: recursively copy keys here? is this needed? we refresh on expand anyway
```

Comment:
todo

Reply:
this isn't a problem anymore as I'm handling the refreshing differently now